Spec/RestClientChannel.swift

```diff
@@ -155,6 +155,54 @@ class RestClientChannel: QuickSpec {
                     expect(mockExecutor.requests.count).to(equal(1))
                 }
             }
+
+            // RSL1g
+            context("Identified clients with a clientId") {
+
+                // RSL1g4
+                pending("when publishing a Message with an explicit clientId that is incompatible with the identified client’s clientId") {
+                    let options = AblyTests.commonAppSetup()
+                    options.clientId = "john"
+                    let client = ARTRest(options: options)
+                    client.httpExecutor = mockExecutor
+                    let channel = client.channels.get("test")
+
+                    mockExecutor.beforeRequest = { request in
+                        if request.URL!.absoluteString.containsString("/channels/test/messages") {
+                            switch extractBodyAsMessages(request) {
+                            case .Failure(let error):
+                                fail(error)
+                            case .Success(let rawMessages):
+                                expect(rawMessages.unbox.first!["clientId"] as? String).to(equal("tester"))
```

Comment:
I'm not sure I get this. The spec says that the library should catch the error, so the request this block is intercepting shouldn't happen, isn't that right?